src/Bicep.Core/Syntax/AccessExpressionSyntax.cs

```diff
@@ -27,4 +27,22 @@ protected AccessExpressionSyntax(SyntaxBase baseExpression, Token? safeAccessMar
     public bool IsSafeAccess => SafeAccessMarker is not null;
 
     public abstract AccessExpressionSyntax AsSafeAccess();
+
+    public abstract string? TryGetPropertyName();
+
+    /// <returns>Returns the base expression chain of this property access syntax. The first element is the left-most expression.
+    /// Example: ext.config.property, ext is first, config is second, etc.</returns>
+    public IReadOnlyList<SyntaxBase> GetBaseExpressionChain()
+    {
+        var currentBaseExpression = BaseExpression;
+        var baseExpressions = new List<SyntaxBase>();
+
+        while (currentBaseExpression is not null)
+        {
+            baseExpressions.Insert(0, currentBaseExpression);
```

Comment:
Insert(0, ...) is O(n). It would be better to append the list and return baseExpressions.Reverse().